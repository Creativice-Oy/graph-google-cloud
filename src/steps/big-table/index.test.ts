import {
  createMockStepExecutionContext,
  Recording,
} from '@jupiterone/integration-sdk-testing';
import {
  fetchAppProfiles,
  fetchBackups,
  fetchClusters,
  fetchInstances,
  fetchOperations,
  fetchTables,
} from '.';
import { integrationConfig } from '../../../test/config';
import { setupGoogleCloudRecording } from '../../../test/recording';
import { IntegrationConfig } from '../../types';
import {
  bigTableEntities,
  RELATIONSHIP_TYPE_CLUSTER_HAS_BACKUP,
  RELATIONSHIP_TYPE_INSTANCE_HAS_APP_PROFILE,
  RELATIONSHIP_TYPE_INSTANCE_HAS_CLUSTER,
  RELATIONSHIP_TYPE_INSTANCE_HAS_TABLE,
} from './constants';

describe('#fetchOperations', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupGoogleCloudRecording({
      directory: __dirname,
      name: 'fetchOperations',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should collect data', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: integrationConfig,
    });

    await fetchOperations(context);

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      numCollectedRelationships: context.jobState.collectedRelationships.length,
      collectedEntities: context.jobState.collectedEntities,
      collectedRelationships: context.jobState.collectedRelationships,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    expect(
      context.jobState.collectedEntities.filter(
        (e) => e.type === bigTableEntities.OPERATIONS._type,
      ),
    ).toMatchGraphObjectSchema({
      _class: ['Task'],
      schema: {
        additionalProperties: false,
        properties: {
          _type: { const: 'google_bigtable_operation' },
          name: { type: 'string' },
          projectId: { type: 'string' },
          done: { type: 'boolean' },
        },
      },
    });
  });
});

describe('#fetchInstances', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupGoogleCloudRecording({
      directory: __dirname,
      name: 'fetchInstances',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should collect data', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: integrationConfig,
    });

    await fetchInstances(context);

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      numCollectedRelationships: context.jobState.collectedRelationships.length,
      collectedEntities: context.jobState.collectedEntities,
      collectedRelationships: context.jobState.collectedRelationships,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    expect(
      context.jobState.collectedEntities.filter(
        (e) => e.type === bigTableEntities.INSTANCES._type,
      ),
    ).toMatchGraphObjectSchema({
      _class: ['Database'],
      schema: {
        additionalProperties: false,
        properties: {
          _type: { const: 'google_bigtable_instance' },
          name: { type: 'string' },
          projectId: { type: 'string' },
          state: { type: 'string' },
          type: { type: 'string' },
        },
      },
    });
  });
});

describe('#fetchAppProfiles', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupGoogleCloudRecording({
      directory: __dirname,
      name: 'fetchAppProfiles',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should collect data', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: integrationConfig,
    });

    await fetchAppProfiles(context);

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      numCollectedRelationships: context.jobState.collectedRelationships.length,
      collectedEntities: context.jobState.collectedEntities,
      collectedRelationships: context.jobState.collectedRelationships,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    expect(
      context.jobState.collectedEntities.filter(
        (e) => e.type === bigTableEntities.APP_PROFILES._type,
      ),
    ).toMatchGraphObjectSchema({
      _class: ['Configuration'],
      schema: {
        additionalProperties: false,
        properties: {
          _type: { const: 'google_bigtable_app_profile' },
          name: { type: 'string' },
          instanceId: { type: 'string' },
          projectId: { type: 'string' },
          etag: { type: 'string' },
          description: { type: 'string' },
        },
      },
    });

    expect(
      context.jobState.collectedRelationships.filter(
        (e) => e._type === RELATIONSHIP_TYPE_INSTANCE_HAS_APP_PROFILE,
      ),
    ).toMatchDirectRelationshipSchema({
      schema: {
        properties: {
          _class: { const: 'HAS' },
          _type: {
            const: 'google_bigtable_instance_has_app_profile',
          },
        },
      },
    });
  });
});

describe('#fetchClusters', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupGoogleCloudRecording({
      directory: __dirname,
      name: 'fetchClusters',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should collect data', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: integrationConfig,
    });

    await fetchClusters(context);

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      numCollectedRelationships: context.jobState.collectedRelationships.length,
      collectedEntities: context.jobState.collectedEntities,
      collectedRelationships: context.jobState.collectedRelationships,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    expect(
      context.jobState.collectedEntities.filter(
        (e) => e.type === bigTableEntities.CLUSTERS._type,
      ),
    ).toMatchGraphObjectSchema({
      _class: ['Cluster'],
      schema: {
        additionalProperties: false,
        properties: {
          _type: { const: 'google_bigtable_cluster' },
          name: { type: 'string' },
          projectId: { type: 'string' },
          instanceId: { type: 'string' },
          state: { type: 'string' },
          location: { type: 'string' },
          defaultStorageType: { type: 'string' },
          serveNodes: { type: 'number' },
          kmsKeyName: { type: 'string' },
        },
      },
    });

    expect(
      context.jobState.collectedRelationships.filter(
        (e) => e._type === RELATIONSHIP_TYPE_INSTANCE_HAS_CLUSTER,
      ),
    ).toMatchDirectRelationshipSchema({
      schema: {
        properties: {
          _class: { const: 'HAS' },
          _type: {
            const: 'google_bigtable_instance_has_cluster',
          },
        },
      },
    });
  });
});

describe('#fetchBackups', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupGoogleCloudRecording({
      directory: __dirname,
      name: 'fetchBackups',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should collect data', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: integrationConfig,
    });

    await fetchBackups(context);

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      numCollectedRelationships: context.jobState.collectedRelationships.length,
      collectedEntities: context.jobState.collectedEntities,
      collectedRelationships: context.jobState.collectedRelationships,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    expect(
      context.jobState.collectedEntities.filter(
        (e) => e.type === bigTableEntities.BACKUPS._type,
      ),
    ).toMatchGraphObjectSchema({
      _class: ['Backup'],
      schema: {
        additionalProperties: false,
        properties: {
          _type: { const: 'google_bigtable_backup' },
          name: { type: 'string' },
          projectId: { type: 'string' },
          instanceId: { type: 'string' },
          clusterId: { type: 'string' },
          sourceTable: { type: 'string' },
          expireTime: { type: 'string' },
          startTime: { type: 'string' },
          endTime: { type: 'string' },
          sizeBytes: { type: 'string' },
          state: { type: 'string' },
          encryptionType: { type: 'string' },
          kmsKeyVersion: { type: 'string' },
        },
      },
    });

    expect(
      context.jobState.collectedRelationships.filter(
        (e) => e._type === RELATIONSHIP_TYPE_CLUSTER_HAS_BACKUP,
      ),
    ).toMatchDirectRelationshipSchema({
      schema: {
        properties: {
          _class: { const: 'HAS' },
          _type: {
            const: 'google_bigtable_cluster_has_backup',
          },
        },
      },
    });
  });
});

describe('#fetchTables', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupGoogleCloudRecording({
      directory: __dirname,
      name: 'fetchTables',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should collect data', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: integrationConfig,
    });

    await fetchTables(context);

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      numCollectedRelationships: context.jobState.collectedRelationships.length,
      collectedEntities: context.jobState.collectedEntities,
      collectedRelationships: context.jobState.collectedRelationships,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    expect(
      context.jobState.collectedEntities.filter(
        (e) => e.type === bigTableEntities.TABLES._type,
      ),
    ).toMatchGraphObjectSchema({
      _class: ['DataCollection'],
      schema: {
        additionalProperties: false,
        properties: {
          _type: { const: 'google_bigtable_table' },
          name: { type: 'string' },
          projectId: { type: 'string' },
          instanceId: { type: 'string' },
          granularity: { type: 'string' },
          backup: { type: 'string' },
        },
      },
    });

    expect(
      context.jobState.collectedRelationships.filter(
        (e) => e._type === RELATIONSHIP_TYPE_CLUSTER_HAS_BACKUP,
      ),
    ).toMatchDirectRelationshipSchema({
      schema: {
        properties: {
          _class: { const: 'HAS' },
          _type: {
            const: 'google_bigtable_instance_has_table',
          },
        },
      },
    });
  });
});
