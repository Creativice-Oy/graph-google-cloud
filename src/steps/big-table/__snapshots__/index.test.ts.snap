// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#fetchAppProfiles should collect data 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "Database",
      ],
      "_key": "bigtable_instance:projects/j1-gc-integration-dev-v3/instances/bt-instance",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "displayName": "bt-instance",
            "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance",
            "state": "READY",
            "type": "PRODUCTION",
          },
        },
      ],
      "_type": "google_bigtable_instance",
      "active": true,
      "createdOn": undefined,
      "displayName": "bt-instance",
      "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance",
      "state": "READY",
      "type": "PRODUCTION",
      "webLink": "https://console.cloud.google.com/bigtable/instances/bt-instance/overview?project=j1-gc-integration-dev-v3",
    },
    Object {
      "_class": Array [
        "Configuration",
      ],
      "_key": "bigtable_appProfile:projects/j1-gc-integration-dev-v3/instances/bt-instance/appProfiles/bt-profile",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance/appProfiles/bt-profile",
            "singleClusterRouting": Object {
              "allowTransactionalWrites": true,
              "clusterId": "bt-instance",
            },
          },
        },
      ],
      "_type": "google_bigtable_app_profile",
      "createdOn": undefined,
      "description": undefined,
      "displayName": "projects/j1-gc-integration-dev-v3/instances/bt-instance/appProfiles/bt-profile",
      "etag": undefined,
      "instanceId": "projects/j1-gc-integration-dev-v3/instances/bt-instance",
      "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance/appProfiles/bt-profile",
      "singleClusterRouting.allowTransactionalWrites": true,
      "singleClusterRouting.clusterId": "bt-instance",
      "webLink": "https://console.cloud.google.com/bigtable/instances/bt-instance/app-profiles/bt-profile?project=j1-gc-integration-dev-v3",
    },
    Object {
      "_class": Array [
        "Configuration",
      ],
      "_key": "bigtable_appProfile:projects/j1-gc-integration-dev-v3/instances/bt-instance/appProfiles/default",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "description": "Default application profile for this instance. This profile is used if you do not supply a different profile ID at connection time.",
            "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance/appProfiles/default",
            "singleClusterRouting": Object {
              "allowTransactionalWrites": true,
              "clusterId": "bt-instance",
            },
          },
        },
      ],
      "_type": "google_bigtable_app_profile",
      "createdOn": undefined,
      "description": "Default application profile for this instance. This profile is used if you do not supply a different profile ID at connection time.",
      "displayName": "projects/j1-gc-integration-dev-v3/instances/bt-instance/appProfiles/default",
      "etag": undefined,
      "instanceId": "projects/j1-gc-integration-dev-v3/instances/bt-instance",
      "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance/appProfiles/default",
      "singleClusterRouting.allowTransactionalWrites": true,
      "singleClusterRouting.clusterId": "bt-instance",
      "webLink": "https://console.cloud.google.com/bigtable/instances/bt-instance/app-profiles/default?project=j1-gc-integration-dev-v3",
    },
  ],
  "collectedRelationships": Array [
    Object {
      "_class": "HAS",
      "_fromEntityKey": "bigtable_instance:projects/j1-gc-integration-dev-v3/instances/bt-instance",
      "_key": "bigtable_instance:projects/j1-gc-integration-dev-v3/instances/bt-instance|has|bigtable_appProfile:projects/j1-gc-integration-dev-v3/instances/bt-instance/appProfiles/bt-profile",
      "_toEntityKey": "bigtable_appProfile:projects/j1-gc-integration-dev-v3/instances/bt-instance/appProfiles/bt-profile",
      "_type": "google_bigtable_instance_has_app_profile",
      "displayName": "HAS",
    },
    Object {
      "_class": "HAS",
      "_fromEntityKey": "bigtable_instance:projects/j1-gc-integration-dev-v3/instances/bt-instance",
      "_key": "bigtable_instance:projects/j1-gc-integration-dev-v3/instances/bt-instance|has|bigtable_appProfile:projects/j1-gc-integration-dev-v3/instances/bt-instance/appProfiles/default",
      "_toEntityKey": "bigtable_appProfile:projects/j1-gc-integration-dev-v3/instances/bt-instance/appProfiles/default",
      "_type": "google_bigtable_instance_has_app_profile",
      "displayName": "HAS",
    },
  ],
  "encounteredTypes": Array [
    "google_bigtable_instance",
    "google_bigtable_app_profile",
    "google_bigtable_instance_has_app_profile",
  ],
  "numCollectedEntities": 3,
  "numCollectedRelationships": 2,
}
`;

exports[`#fetchBackups should collect data 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "Database",
      ],
      "_key": "bigtable_instance:projects/j1-gc-integration-dev-v3/instances/bt-instance",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "displayName": "bt-instance",
            "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance",
            "state": "READY",
            "type": "PRODUCTION",
          },
        },
      ],
      "_type": "google_bigtable_instance",
      "active": true,
      "createdOn": undefined,
      "displayName": "bt-instance",
      "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance",
      "state": "READY",
      "type": "PRODUCTION",
      "webLink": "https://console.cloud.google.com/bigtable/instances/bt-instance/overview?project=j1-gc-integration-dev-v3",
    },
    Object {
      "_class": Array [
        "Cluster",
      ],
      "_key": "bigtable_cluster:projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "defaultStorageType": "HDD",
            "location": "projects/j1-gc-integration-dev-v3/locations/us-central1-b",
            "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance",
            "serveNodes": 3,
            "state": "READY",
          },
        },
      ],
      "_type": "google_bigtable_cluster",
      "active": true,
      "createdOn": undefined,
      "defaultStorageType": "HDD",
      "displayName": "projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance",
      "instanceId": "projects/j1-gc-integration-dev-v3/instances/bt-instance",
      "kmsKeyName": undefined,
      "location": "projects/j1-gc-integration-dev-v3/locations/us-central1-b",
      "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance",
      "serveNodes": 3,
      "state": "READY",
      "webLink": "https://console.cloud.google.com/bigtable/instances/bt-instance/overview?project=j1-gc-integration-dev-v3",
    },
    Object {
      "_class": Array [
        "Backup",
      ],
      "_key": "bigtable_backup:projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance/backups/tf-backup",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "encryptionInfo": Object {
              "encryptionType": "GOOGLE_DEFAULT_ENCRYPTION",
            },
            "endTime": "2021-08-12T00:46:40.026785Z",
            "expireTime": "2021-08-13T00:46:37.503Z",
            "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance/backups/tf-backup",
            "sourceTable": "projects/j1-gc-integration-dev-v3/instances/bt-instance/tables/tf-table",
            "startTime": "2021-08-12T00:46:39.326777Z",
            "state": "READY",
          },
        },
      ],
      "_type": "google_bigtable_backup",
      "active": true,
      "clusterId": "projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance",
      "createdOn": undefined,
      "displayName": "projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance/backups/tf-backup",
      "encryptionType": "GOOGLE_DEFAULT_ENCRYPTION",
      "endTime": 1628729200026,
      "expireTime": 1628815597503,
      "instanceId": "projects/j1-gc-integration-dev-v3/instances/bt-instance",
      "kmsKeyVersion": undefined,
      "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance/backups/tf-backup",
      "sizeBytes": undefined,
      "sourceTable": "projects/j1-gc-integration-dev-v3/instances/bt-instance/tables/tf-table",
      "startTime": 1628729199326,
      "state": "READY",
      "webLink": "https://console.cloud.google.com/bigtable/instances/bt-instance/backups?project=j1-gc-integration-dev-v3",
    },
  ],
  "collectedRelationships": Array [
    Object {
      "_class": "HAS",
      "_fromEntityKey": "bigtable_instance:projects/j1-gc-integration-dev-v3/instances/bt-instance",
      "_key": "bigtable_instance:projects/j1-gc-integration-dev-v3/instances/bt-instance|has|bigtable_cluster:projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance",
      "_toEntityKey": "bigtable_cluster:projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance",
      "_type": "google_bigtable_instance_has_cluster",
      "displayName": "HAS",
    },
    Object {
      "_class": "HAS",
      "_fromEntityKey": "bigtable_cluster:projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance",
      "_key": "bigtable_cluster:projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance|has|bigtable_backup:projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance/backups/tf-backup",
      "_toEntityKey": "bigtable_backup:projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance/backups/tf-backup",
      "_type": "google_bigtable_cluster_has_backup",
      "displayName": "HAS",
    },
  ],
  "encounteredTypes": Array [
    "google_bigtable_instance",
    "google_bigtable_cluster",
    "google_bigtable_instance_has_cluster",
    "google_bigtable_backup",
    "google_bigtable_cluster_has_backup",
  ],
  "numCollectedEntities": 3,
  "numCollectedRelationships": 2,
}
`;

exports[`#fetchClusters should collect data 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "Database",
      ],
      "_key": "bigtable_instance:projects/j1-gc-integration-dev-v3/instances/bt-instance",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "displayName": "bt-instance",
            "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance",
            "state": "READY",
            "type": "PRODUCTION",
          },
        },
      ],
      "_type": "google_bigtable_instance",
      "active": true,
      "createdOn": undefined,
      "displayName": "bt-instance",
      "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance",
      "state": "READY",
      "type": "PRODUCTION",
      "webLink": "https://console.cloud.google.com/bigtable/instances/bt-instance/overview?project=j1-gc-integration-dev-v3",
    },
    Object {
      "_class": Array [
        "Cluster",
      ],
      "_key": "bigtable_cluster:projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "defaultStorageType": "HDD",
            "location": "projects/j1-gc-integration-dev-v3/locations/us-central1-b",
            "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance",
            "serveNodes": 3,
            "state": "READY",
          },
        },
      ],
      "_type": "google_bigtable_cluster",
      "active": true,
      "createdOn": undefined,
      "defaultStorageType": "HDD",
      "displayName": "projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance",
      "instanceId": "projects/j1-gc-integration-dev-v3/instances/bt-instance",
      "kmsKeyName": undefined,
      "location": "projects/j1-gc-integration-dev-v3/locations/us-central1-b",
      "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance",
      "serveNodes": 3,
      "state": "READY",
      "webLink": "https://console.cloud.google.com/bigtable/instances/bt-instance/overview?project=j1-gc-integration-dev-v3",
    },
  ],
  "collectedRelationships": Array [
    Object {
      "_class": "HAS",
      "_fromEntityKey": "bigtable_instance:projects/j1-gc-integration-dev-v3/instances/bt-instance",
      "_key": "bigtable_instance:projects/j1-gc-integration-dev-v3/instances/bt-instance|has|bigtable_cluster:projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance",
      "_toEntityKey": "bigtable_cluster:projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance",
      "_type": "google_bigtable_instance_has_cluster",
      "displayName": "HAS",
    },
  ],
  "encounteredTypes": Array [
    "google_bigtable_instance",
    "google_bigtable_cluster",
    "google_bigtable_instance_has_cluster",
  ],
  "numCollectedEntities": 2,
  "numCollectedRelationships": 1,
}
`;

exports[`#fetchInstances should collect data 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "Database",
      ],
      "_key": "bigtable_instance:projects/j1-gc-integration-dev-v3/instances/bt-instance",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "displayName": "bt-instance",
            "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance",
            "state": "READY",
            "type": "PRODUCTION",
          },
        },
      ],
      "_type": "google_bigtable_instance",
      "active": true,
      "createdOn": undefined,
      "displayName": "bt-instance",
      "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance",
      "state": "READY",
      "type": "PRODUCTION",
      "webLink": "https://console.cloud.google.com/bigtable/instances/bt-instance/overview?project=j1-gc-integration-dev-v3",
    },
  ],
  "collectedRelationships": Array [],
  "encounteredTypes": Array [
    "google_bigtable_instance",
  ],
  "numCollectedEntities": 1,
  "numCollectedRelationships": 0,
}
`;

exports[`#fetchOperations should collect data 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "Task",
      ],
      "_key": "bigtable_operation:operations/projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance/backups/tf-backup/locations/us-central1-b/operations/4865267906225648703",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "done": true,
            "metadata": Object {
              "@type": "type.googleapis.com/google.bigtable.admin.v2.CreateBackupMetadata",
              "endTime": "2021-08-12T00:46:40.151706669Z",
              "name": "tf-backup",
              "sourceTable": "projects/j1-gc-integration-dev-v3/instances/bt-instance/tables/tf-table",
              "startTime": "2021-08-12T00:46:39.327216499Z",
            },
            "name": "operations/projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance/backups/tf-backup/locations/us-central1-b/operations/4865267906225648703",
            "response": Object {
              "@type": "type.googleapis.com/google.bigtable.admin.v2.Backup",
              "encryptionInfo": Object {
                "encryptionType": "GOOGLE_DEFAULT_ENCRYPTION",
              },
              "endTime": "2021-08-12T00:46:40.026785Z",
              "expireTime": "2021-08-13T00:46:37.503Z",
              "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance/backups/tf-backup",
              "sourceTable": "projects/j1-gc-integration-dev-v3/instances/bt-instance/tables/tf-table",
              "startTime": "2021-08-12T00:46:39.326777Z",
              "state": "READY",
            },
          },
        },
      ],
      "_type": "google_bigtable_operation",
      "createdOn": undefined,
      "displayName": "operations/projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance/backups/tf-backup/locations/us-central1-b/operations/4865267906225648703",
      "done": true,
      "name": "operations/projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance/backups/tf-backup/locations/us-central1-b/operations/4865267906225648703",
    },
    Object {
      "_class": Array [
        "Task",
      ],
      "_key": "bigtable_operation:operations/projects/j1-gc-integration-dev-v3/instances/bt-instance/locations/us-central1-b/operations/8951918535169126122",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "done": true,
            "metadata": Object {
              "@type": "type.googleapis.com/google.bigtable.admin.v2.CreateInstanceMetadata",
              "finishTime": "2021-08-12T00:40:44.329046107Z",
              "originalRequest": Object {
                "clusters": Object {
                  "bt-instance": Object {
                    "defaultStorageType": "HDD",
                    "location": "projects/j1-gc-integration-dev-v3/locations/us-central1-b",
                    "serveNodes": 3,
                  },
                },
                "instance": Object {
                  "displayName": "bt-instance",
                  "type": "PRODUCTION",
                },
                "instanceId": "bt-instance",
                "parent": "projects/j1-gc-integration-dev-v3",
              },
              "requestTime": "2021-08-12T00:40:43.111693512Z",
            },
            "name": "operations/projects/j1-gc-integration-dev-v3/instances/bt-instance/locations/us-central1-b/operations/8951918535169126122",
            "response": Object {
              "@type": "type.googleapis.com/google.bigtable.admin.v2.Instance",
              "displayName": "bt-instance",
              "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance",
              "state": "READY",
              "type": "PRODUCTION",
            },
          },
        },
      ],
      "_type": "google_bigtable_operation",
      "createdOn": undefined,
      "displayName": "operations/projects/j1-gc-integration-dev-v3/instances/bt-instance/locations/us-central1-b/operations/8951918535169126122",
      "done": true,
      "name": "operations/projects/j1-gc-integration-dev-v3/instances/bt-instance/locations/us-central1-b/operations/8951918535169126122",
    },
    Object {
      "_class": Array [
        "Task",
      ],
      "_key": "bigtable_operation:operations/projects/j1-gc-integration-dev-v3/instances/bt-instance/locations/us-central1-b/operations/5284587272113793734",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "done": true,
            "metadata": Object {
              "@type": "type.googleapis.com/google.bigtable.admin.v2.CreateInstanceMetadata",
              "finishTime": "2021-08-06T17:57:36.880096758Z",
              "originalRequest": Object {
                "clusters": Object {
                  "bt-instance": Object {
                    "defaultStorageType": "HDD",
                    "location": "projects/j1-gc-integration-dev-v3/locations/us-central1-b",
                    "serveNodes": 3,
                  },
                },
                "instance": Object {
                  "displayName": "bt-instance",
                  "type": "PRODUCTION",
                },
                "instanceId": "bt-instance",
                "parent": "projects/j1-gc-integration-dev-v3",
              },
              "requestTime": "2021-08-06T17:57:35.431728193Z",
            },
            "name": "operations/projects/j1-gc-integration-dev-v3/instances/bt-instance/locations/us-central1-b/operations/5284587272113793734",
            "response": Object {
              "@type": "type.googleapis.com/google.bigtable.admin.v2.Instance",
              "displayName": "bt-instance",
              "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance",
              "state": "READY",
              "type": "PRODUCTION",
            },
          },
        },
      ],
      "_type": "google_bigtable_operation",
      "createdOn": undefined,
      "displayName": "operations/projects/j1-gc-integration-dev-v3/instances/bt-instance/locations/us-central1-b/operations/5284587272113793734",
      "done": true,
      "name": "operations/projects/j1-gc-integration-dev-v3/instances/bt-instance/locations/us-central1-b/operations/5284587272113793734",
    },
  ],
  "collectedRelationships": Array [],
  "encounteredTypes": Array [
    "google_bigtable_operation",
  ],
  "numCollectedEntities": 3,
  "numCollectedRelationships": 0,
}
`;

exports[`#fetchTables should collect data 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "Database",
      ],
      "_key": "bigtable_instance:projects/j1-gc-integration-dev-v3/instances/bt-instance",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "displayName": "bt-instance",
            "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance",
            "state": "READY",
            "type": "PRODUCTION",
          },
        },
      ],
      "_type": "google_bigtable_instance",
      "active": true,
      "createdOn": undefined,
      "displayName": "bt-instance",
      "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance",
      "state": "READY",
      "type": "PRODUCTION",
      "webLink": "https://console.cloud.google.com/bigtable/instances/bt-instance/overview?project=j1-gc-integration-dev-v3",
    },
    Object {
      "_class": Array [
        "DataCollection",
      ],
      "_key": "bigtable_table:projects/j1-gc-integration-dev-v3/instances/bt-instance/tables/tf-table",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance/tables/tf-table",
          },
        },
      ],
      "_type": "google_bigtable_table",
      "backup": undefined,
      "classification": null,
      "createdOn": undefined,
      "displayName": "projects/j1-gc-integration-dev-v3/instances/bt-instance/tables/tf-table",
      "granularity": undefined,
      "instanceId": "projects/j1-gc-integration-dev-v3/instances/bt-instance",
      "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance/tables/tf-table",
      "sourceType": undefined,
      "webLink": "https://console.cloud.google.com/bigtable/instances/bt-instance/tables?project=j1-gc-integration-dev-v3",
    },
  ],
  "collectedRelationships": Array [
    Object {
      "_class": "HAS",
      "_fromEntityKey": "bigtable_instance:projects/j1-gc-integration-dev-v3/instances/bt-instance",
      "_key": "bigtable_instance:projects/j1-gc-integration-dev-v3/instances/bt-instance|has|bigtable_table:projects/j1-gc-integration-dev-v3/instances/bt-instance/tables/tf-table",
      "_toEntityKey": "bigtable_table:projects/j1-gc-integration-dev-v3/instances/bt-instance/tables/tf-table",
      "_type": "google_bigtable_instance_has_table",
      "displayName": "HAS",
    },
  ],
  "encounteredTypes": Array [
    "google_bigtable_instance",
    "google_bigtable_table",
    "google_bigtable_instance_has_table",
  ],
  "numCollectedEntities": 2,
  "numCollectedRelationships": 1,
}
`;
